src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -1108,7 +1108,7 @@ public HttpMessage handleApiOther(HttpMessage msg, String name,
 				logger.error(e.getMessage(), e);
 			}
 			return msg;
-		} else if (OTHER_SET_PROXY.equals(name)) {
+		} else if (OTHER_SET_PROXY.equals(name) && params != null) {
```

Comment:
This is a FP (from `handleShortcut` and `API` `params` is always non-null), it would be better if it was ignored/suppressed in LGTM, is that possible?

Reply:
If you look at lgtm it does actually show a couple of examples where it could be null, eg https://lgtm.com/projects/g/zaproxy/zaproxy/snapshot/dist-1984191023-1534248079015/files/src/org/zaproxy/zap/extension/api/CoreAPI.java?sort=name&dir=ASC&mode=heatmap&showExcluded=false#x9e41f5cf6071ea73:1 and https://lgtm.com/projects/g/zaproxy/zaproxy/snapshot/dist-1984191023-1534248079015/files/src/org/zaproxy/zap/extension/api/CoreAPI.java?sort=name&dir=ASC&mode=heatmap&showExcluded=false#x9038eb4ad7849464:1

They shouldnt apply in this case, but we can either cope with null, change those calls or mark as an FP in lgtm - I chose the first but happy to go with the consensus ..

Reply:
I'm fine with changing the other calls to pass a non-null object, that's safer overall, otherwise I'd prefer ignore those (e.g. why just a null check in this case? why not add null checks in other places where `params` is being accessed?).